front/lib/resources/data_source_resource.ts

```diff
@@ -42,13 +42,38 @@ export type FetchDataSourceOrigin =
   | "cli_delete_document"
   | "vault_patch_content"
   | "data_source_view_create"
-  | "poke_data_source_config"
+  | "poke_data_sources"
+  | "poke_data_sources_search"
+  | "poke_data_sources_config"
+  | "poke_data_sources_documents"
+  | "poke_data_sources_permissions"
   | "registry_lookup"
   | "data_source_get_or_post"
   | "data_source_managed_update"
   | "vault_data_source_config"
   | "vault_patch_or_delete_data_source"
-  | "vault_data_source_documents";
+  | "vault_data_source_documents"
```

Comment:
Unrelated to this PR; seems a bit surprising to have to create string types for each call site, and track teh origin this way. I wasn't part of the design of this so maybe it's legit but it leaves me wondering; is it the right way to trace calls of resources?

Reply:
This is transitory to keep track of usage of [name] across the entire product and will get killed once everything is removed